### PR TITLE
fix: [SPG-2082]: K8sCanaryDeleteServiceElement variable change

### PR DESCRIPTION
**Pull request by TarunUba1**
## Describe your changes
Only setting K8sCanaryDeleteServiceElement when export manifest is not set to be true. As the variable was being set for even dry run, a consecutive deployment of canary was failing as that variable was already set

## Testing
1. Test with export and import steps
Screenshot 2023-02-22 at 2 41 10 PM
Screenshot 2023-02-22 at 2 40 52 PM
2. Test with export and nothing
Screenshot 2023-02-22 at 2 38 22 PM
Screenshot 2023-02-22 at 2 38 10 PM

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-ALL: `trigger tiAll`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44202)

### CODE REVIEW

File: 400-rest/src/main/java/software/wings/sm/states/k8s/K8sCanaryDeploy.java [hunk 1]
```diff
@@ -199,7 +199,9 @@ public ExecutionResponse executeK8sTask(ExecutionContext context, String activit
                 featureFlagService.isEnabled(CDP_USE_K8S_DECLARATIVE_ROLLBACK, infraMapping.getAccountId()))
             .build();
     ExecutionResponse response = queueK8sDelegateTask(context, k8sTaskParameters, appManifestMap);
-    saveK8sCanaryDeployRun(context);
+    if (!exportManifests) {
```
Comment (abosii-harness) [inline 400-rest/src/main/java/software/wings/sm/states/k8s/K8sCanaryDeploy.java:202]: Let's add a test for this use case

**Review by abosii-harness — CHANGES_REQUESTED** [review]

File: 400-rest/src/main/java/software/wings/sm/states/k8s/K8sCanaryDeploy.java (hunk 1 above)
Comment (TarunUba1) [reply 400-rest/src/main/java/software/wings/sm/states/k8s/K8sCanaryDeploy.java:202]: Done

**Review by abosii-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r1, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44202)* status: :shipit: complete! ✓

**Review by abhinavsingh1196 — APPROVED** [review]
> Reviewed 1 of 1 files at r1, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44202)* status: :shipit: complete! ✓

**Pull request merged**